### ISSUE TITLE
💅 incorrect `noMisleadingCharacterClass` on regexp

### ISSUE THREAD

**Issue by piranha** [biomejs/biome#4950]
### Environment information

```bash
CLI:
  Version:                      1.9.4
```

### Rule name

noMisleadingCharacterClass

### Playground link

https://biomejs.dev/playground/?code=JwB0AGUAcwB0ACAAXAB1ADAAMwAwADEAIAB0AGgAaQBzACcALgByAGUAcABsAGEAYwBlACgALwBbAFwAdQAwADMAMAAwAC0AXAB1ADAAMwAwADIAXQAvAGcALAAgACcAWABYAFgAJwApAA%3D%3D

### Expected result

I don't think `/[\u0300-\u0302]/` warrants this error, it seems to be working well (it did work for me way before I installed biome 😁).

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by bushuai** [comment]
Hey @arendjr I'd like to take this issue. :D

**Comment by Conaclos** [comment]
Maybe it is working because of Unicode escape sequences?

**Comment by arendjr** [comment]
@Conaclos The rule is intended to flag characters that contain multiple code points, which doesn't have expected results because the code points will be treated separately by the character class. But the example given here is a range over _individual_ code points. This usage isn't really ambiguous or otherwise confusing, so it shouldn't trigger the rule.
